src/Classes/TradeQueryRequests.lua

```diff
@@ -0,0 +1,275 @@
+		end
+		if params.callbackQueryId then
+			params.callbackQueryId(queryId)
+		end
+		self:FetchResults(itemHashes, queryId, callback)
+	end)
+end
+
+---Perform search and run callback function on returned item hashes.
+---Item info has to be fetched seperately 
+---@param league string
+---@param query string
+---@param callback fun(itemHashes:string[], queryId:string, errMsg:string)
+function TradeQueryRequestsClass:PerformSearch(league, query, callback)
+	table.insert(self.requestQueue["search"], {
+		url = "https://www.pathofexile.com/api/trade/search/"..league,
+		body = query,
+		callback = function(response, errMsg)
+			if errMsg then
+				return callback(nil, nil, errMsg)
+			end
+			local response = dkjson.decode(response)
+			if not response then
+				errMsg =  "Failed to Get Trade response"
+				return callback(nil, nil, errMsg)
+			end
+			if not response.result or #response.result == 0 then
+				if response.error then
+					if response.error.code == 2 then
+						errMsg = "Complex Query - Please provide your POESESSID"
+					elseif response.error.message then
+						errMsg = response.error.message
+					end
+				else
+					errMsg = "No Matching Results Found"
+				end
+				return callback(nil, nil, errMsg)
+			end
+			callback(response.result, response.id, errMsg)
+		end,
+	})
+end
+
+---Fetch item details for itemHashes
+---@param itemHashes string[]
+---@param queryId string
+---@param callback fun(items:table, errMsg:string)
+function TradeQueryRequestsClass:FetchResults(itemHashes, queryId, callback)
+	local quantity_found = math.min(#itemHashes, self.maxFetchPerSearch)
+	local max_block_size = 10
+	local items = {}
+	for fetch_block_start = 1, quantity_found, max_block_size do
+		local fetch_block_end = math.min(fetch_block_start + max_block_size - 1, quantity_found)
+		local param_item_hashes = table.concat({unpack(itemHashes, fetch_block_start, fetch_block_end)}, ",")
+		local fetch_url = "https://www.pathofexile.com/api/trade/fetch/"..param_item_hashes.."?query="..queryId
+		self:FetchResultBlock(fetch_url, function(itemBlock, errMsg)
+			if errMsg then
+				return callback(nil, errMsg)
+			end
+			for _, item in pairs(itemBlock) do
+				table.insert(items, item)
+			end
+			-- finished fetching item blocks
+			if #items == quantity_found then
+				callback(items)
+			end
+		end)
+	end
+end
+
+---Fetch details for paginated items
+---@param url string
+---@param callback fun(items: table, errMsg:string)
+function TradeQueryRequestsClass:FetchResultBlock(url, callback)
+	table.insert(self.requestQueue["fetch"], {
+		url = url,
+		callback = function(response, errMsg)
+			if errMsg then
+				return callback(nil, errMsg)
+			end
+			local response, response_err = dkjson.decode(response)
+			if not response or not response.result then
+				local errMsg
+				if response_err then
+					errMsg = "JSON Parse Error "
+				else
+					errMsg = "Failed to Get Trade Items"
+				end
+				return callback(nil, errMsg)
```

Comment:
This is clobbering the existing errorMsg if there was one.  Consider setting `errMsg = "JSON Parse Error: " + errMsg` without redeclaring `errMsg`